gui/src/ts/components/benchmark/BenchmarkAnswerPage.tsx

```diff
@@ -38,6 +56,251 @@ const QuestionAnswerChoiceCard: React.FunctionComponent<{
+
+          // Extract nodes from path [node link node link node]
+          for (let i = 0; i < path.length; i += 2) {
+            const nodeId = path[i];
+            if (!nodes[nodeId]) {
+              nodes[nodeId] = {
+                id: nodeId,
+                label: nodeId,
+                incomingEdges: 0,
+                outgoingEdges: 0,
+                paths: [],
+              };
+            }
+          }
+
+          // Extract links from path [node link node link node]
+          for (let i = 1; i < path.length; i += 2) {
+            const sourceNodeId = path[i - 1];
+            const targetNodeId = path[i + 1];
+
+            const linkId = `${pathId}-${i}`;
+            const pathInfo = {
+              questionAnswerPathId,
+              choiceAnalysisId,
+              score,
+            };
+            // Add path info to link
+            links[linkId] = {
+              source: sourceNodeId,
+              target: targetNodeId,
+              sourceId: sourceNodeId,
+              targetId: targetNodeId,
+              id: linkId,
+              pathId,
+              label: path[i],
+              ...pathInfo,
+            };
+
+            const sourceNode = nodes[sourceNodeId];
+            const targetNode = nodes[targetNodeId];
+
+            // Increment node edge counts
+            sourceNode.outgoingEdges += 1;
+            targetNode.incomingEdges += 1;
+
+            // Add path info to node
+            if (!sourceNode.paths.some((path) => path.id === pathId)) {
+              sourceNode.paths.push({
+                id: pathId,
+                ...pathInfo,
+              });
+            }
+            if (!targetNode.paths.some((path) => path.id === pathId)) {
+              targetNode.paths.push({
+                id: pathId,
+                ...pathInfo,
+              });
+            }
+          }
+        });
+      });
+    });
+
+  // Sort paths of each node by score
+  for (const node of Object.values(nodes)) {
+    node.paths = node.paths.sort((path1, path2) => path1.score - path2.score);
+  }
+
+  return {nodes, links};
+};
+
+const answerExplanationGraphStyles = makeStyles({
+  graphLegendContainer: {
+    display: "flex",
+    flexDirection: "row",
+    padding: 0,
+    whiteSpace: "nowrap",
+  },
+});
+
+const AnswerExplanationGraphLegendNode: React.FunctionComponent<{
+  radius?: number;
+  color?: string;
+  opacity?: number;
+}> = ({radius: propRadius, color: propColor, opacity: propOpacity}) => {
+  const radius = propRadius || 20;
+  const backgroundColor = propColor || "#999";
+  const opacity = propOpacity || 1;
+  return (
+    <ListItemAvatar>
+      <div
+        style={{
+          height: radius + "px",
+          width: radius + "px",
+          borderRadius: "50%",
+          backgroundColor,
+          display: "inline-block",
+          opacity,
+        }}
+      />
+    </ListItemAvatar>
+  );
+};
+
+const AnswerExplanationGraph: React.FunctionComponent<{
```

Comment:
This file is pretty big, can you break out these components? (and prefix with Benchmark i.e., BenchmarkAnswerExplanationGraph).

Reply:
extracted BenchmarkAnswerExplanationGraph